doc/DatabaseOperations.md

```diff
@@ -0,0 +1,154 @@
+# Postgres Database Components and API
+
+Accessing the database involves setting up Postgres tables according to a table
+definition, and adding, retrieving, and modifying the data within those tables.
+A fluid component has been defined for performing these operations using
+JavaScript.
+
+The implementation is a combination of [PostgreSQL](https://www.postgresql.org/),
+[node-postgres](https://node-postgres.com/), and code to handle SQL and JSON.
+Requests made to the database that manipulate the tables and their contents are
+given in the form of PostgreSQL strings.  Responses from the database are JSON.
+
+Most of the functions return a `Promise`.  The operations described here use
+only `then()`, `resolve()`, and `reject()` functions, allowing the use of
+[Infusion's promise API](https://docs.fluidproject.org/infusion/development/promisesapi).
+
+## APIs
+
+The library consists of a single fluid component that establishes a
+connection to the data base (host, port, etc.) and provides an interface to
+make PostgresSQL queries
+
+### `fluid.postgresdb.request`
+
+The request component is for initializing the connection with the database and
+provides a means to make queries of it.
+
+#### Component options
+
+These options are all set to `null` in the base grade.  Integrators need to set
+these as appropriate to their situation.
+
+| Option            | Type       | Description | Default |
+| ----------------- | ---------- | ----------- | ------- |
+| `databaseName`    | String     | Required. Name of the database containing the tables, e.g., `"fluid_prefsdb"` | `null` |
+| `host`            | String     | Required. The host for the requests, e.g., `"localhost"` | `null` |
+| `port`            | Integer    | Required. The port for the requests, e.g., `5432` | `null` |
+| `user`            | String     | Required. User administrator's name, e.g., `"admin"`| `null` |
+| `password`        | String     | Required for secure implementations. User administrator's password | `null` |
+
+#### Component members
+
+| Member            | Type       | Description |
+| ----------------- | ---------- | ----------- |
+| `pool`            | Object     | Access to the node-postgres instance.
+
+When initialized, the request component allocates a node-postgress [`Pool`](https://node-postgres.com/api/pool)
+and uses it to establish the connection to the database as per the request's
+options.  It is then ready for database queries.
+
+#### Operations API
+
+##### `query(queryString)` (Invoker)
```

Comment:
The name `query` gives an impression of handling `SELECT` statement. Since this function is used for executing all kind of SQL statements. Do you think if it's better to call it `executeSQL()`?

Reply:
The name is from the node-postgres library itself.  Its `query()` method is used for any SQL statement, not just SELECT.  Around the third time that I had typed `that.pool.query()`, I decided that a wrapper for it would simplify the situation; hence `that.query()`.

With the move to using JavaScript classes, the method is now inherited.  However, since I think it's a good idea to attach a rejection handler for logging errors, I've renamed the wrapper to `runSql()`.